sql/2019/09_Accessibility/09_06.sql

```diff
@@ -0,0 +1,32 @@
+/*
+09_06
+10TB
+only tested on the sample data set
+
+extracts all id="*" fields from every website.  
+flattens the dataset, and then counts for duplicates.
+The final result shows how many IDs appear counter times on the page
+using the sample_data.response_bodies_mobile_10k
+there are 777,252 ids that occur just once
+           51,450 ids that appear twice (on one page)
+           13,978 appear 3x
+           and so on
+           
+           a googlead ID appears 5560 times on 3 different sides
+
+*/
+
+select counter, count (*) from(
+
+select url, flat_ids , count(flat_ids) counter from(
+select
+url, regexp_extract_all(body,r'(id=".*?")') ids
```

Comment:
Some nitpicks with this regex:

- case sensitive
- assumes double quotes, need to handle single quotes or none at all
- prefer something like `[^\s'"]*` over `.*` for HTML attribute values to handle boundaries

Reply:
improved regex :)